src/view/pages/Farmer/tabs/Info/index.tsx

```diff
@@ -0,0 +1,217 @@
+ */
+const modes = {
+  edit: 'edit',
+  view: 'view',
+};
+
+/**
+ * Container for application
+ */
+class FarmerInformation extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: OwnPropsType) {
+    super(props);
+    this.state = { 
+      mode: 'view',
+      selectedPaymentCycle: 'Weekly',
+      selectedPaymentMethod: 'Mobile',
+    };
+  }
+
+  private changeMode = (newMode: string) => {
+    this.setState(state => ({ mode: newMode }));
+  }
+
+  private updatePaymentCycle = (value: string) => {
+    this.setState(state => ({ selectedPaymentCycle: value }));
+  }
+  
+  private updatePaymentMethod = (value: string) => {
+    this.setState(state => ({ selectedPaymentMethod: value }));
+  }
+
+  /**
+   * Render method for Farmer Information
+   */
+  public render() {
+    if (this.state.mode === modes.view) {
+      return (
+        <Content padder>
+          <Grid>
+            <Row style={styles.farmerName}>
+              <H1>{this.props.farmerFirstName} {this.props.farmerLastName}</H1>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Phone Number</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerPhoneNumber}</H2>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Business Name</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerBusinessName}</H2>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Col>
+                <Label>Payment Cycle</Label>
+              </Col> 
+              <Col>
+                <Label>Payment Method</Label>
+              </Col>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Col>
+                <H2>{this.state.selectedPaymentCycle}</H2>
+              </Col> 
+              <Col>
+                <H2>{this.state.selectedPaymentMethod}</H2>
+              </Col>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <Label>Notes</Label>
+            </Row>
+            <Row style={styles.infoLabel}>
+              <H2>{this.props.farmerNotes}</H2>
+            </Row>
+            <Row style={styles.editButton}>
+              <Col>
+                <Button block danger onPress={() => this.changeMode(modes.edit)}>
+                  <Text style={styles.buttonText}>Edit</Text>
+                </Button>
+              </Col>
+            </Row>
+          </Grid>
+        </Content>
+      );
+    } else if (this.state.mode === modes.edit) {
+      return (
+        <Content padder>
+          <Grid>
+            <Row style={styles.infoLabel}>
+              <Col>
+                <Label>First Name</Label>
+                </Col>
+                <Col>
+                <Label>Last Name</Label>
+              </Col>
+            </Row>
+            <Row style={styles.input}>
```

Comment:
See the following from Nick's branch:
```
  public render(): JSX.Element {
    return (
      <Content style={style.background} >
        <List
          dataArray={this.searchList()}
          renderRow={this.renderItem}
        />
      </Content>
    );
  }

  /** Function to render the individual list items */
  private renderItem(info: Farmer) {
    return (
      <ListItem
        key={info.id}
        onPress={this.onItemClick}
      >
        <View>
          <Text style={style.name}>
            {info.name}
          </Text>
          <Text style={style.phone}>
            Phone: {info.phoneNumber}
          </Text>
        </View>
      </ListItem>
    );
```

He calls a function on an array of data that specifies how to render each entry of the data. This is not the way you would typically do it however, as most elements do not specifically have a `renderRow` prop or anything similar. Typically this would be done via the following:
```
public render(): JSX.Element {
    return (
      <Content style={style.background} >
        <List>
          {this.searchList().map(this.renderItem)}
        </List>
      </Content>
    );
  }
```

Which is the same thing Nick did, which is why I'm not sure why `native-base` even provides a different API when you can do it with React itself. I would recommend trying to adopt a similar pattern to reduce repeated code if possible

Reply:
This is very interesting, and I totally see your point. This reduces copied code and is much cleaner. However, I'll open an issue for this to go back and do it later since we have to prioritize other things in the time being. Thanks for the insight!

Reply:
That's good to know. Thanks for pointing it out brad.